File: translated_tests/1 Single class test/OnlyStatic.c
#include"OnlyStatic.h"

// instance fields of the class
struct OnlyStatic {
};

// static fields of the class
int static_fieldInitializedAutomatically;
int static_fieldInitializedBeforeConstructor;
int static_privateField;
int static_protectedField;
int static_standardField;
int static_publicField;
int static_counter;

OnlyInstance createOnlyInstance(int privateField, int protectedField, int standardField, int publicField){
		return new_OnlyInstance(privateField, protectedField, standardField, publicField);
	}

void setCounter(int counter){
		static_counter = counter;
	}

void resetCounter(){
		setCounter(0);
	}

int getFieldInitializedAutomatically(){
		return static_fieldInitializedAutomatically;
	}

int getFieldInitializedBeforeConstructor(){
		return static_fieldInitializedBeforeConstructor;
	}

int getPrivateField(){
		return static_privateField;
	}

int getProtectedField(){
		return static_protectedField;
	}

int getStandardField(){
		return static_standardField;
	}

int getPublicField(){
		return static_publicField;
	}

void testContextFields(){
		printf("%s\n", static_privateField);
		printf("%s\n", static_protectedField);
		printf("%s\n", static_standardField);
		printf("%s\n", static_publicField);
	}

void testNoContextFields(){
		printf("%s\n", privateField);
		printf("%s\n", protectedField);
		printf("%s\n", standardField);
		printf("%s\n", publicField);
	}

void testPrivateMethod(){
		printf("Private\n");
	}

void testProtectedMethod(){
		printf("Protected\n");
	}

void testStandardMethod(OnlyStatic this){
		printf("Standard\n");
	}

void testPublicMethod(){
		printf("Public\n");
	}

void test(){
		// 0 0 0 0
		printf("%s\n", getPrivateField());
		printf("%s\n", getProtectedField());
		printf("%s\n", getStandardField());
		printf("%s\n", getPublicField());
		// 0 0 0 0
		printf("%s\n", static_privateField);
		printf("%s\n", static_protectedField);
		printf("%s\n", static_standardField);
		printf("%s\n", static_publicField);

		// 0 0 0 0
		testContextFields();
		// 0 0 0 0
		testNoContextFields();

		// Private Protected Standard Public
		testPrivateMethod();
		testProtectedMethod();
		testStandardMethod();
		testPublicMethod();
	}

File: tests/OnlyStatic.java
public class OnlyStatic {

	private static int fieldInitializedAutomatically;
	private static int fieldInitializedBeforeConstructor = 42;
	private static int privateField;
	protected static int protectedField;
	static int standardField;
	public static int publicField;

	private static int counter;

	public static OnlyInstance createOnlyInstance(int privateField, int protectedField, int standardField, int publicField) {
		return new OnlyInstance(privateField, protectedField, standardField, publicField);
	}

	public static void setCounter(int counter) {
		OnlyStatic.counter = counter;
	}

	public static void resetCounter() {
		OnlyStatic.setCounter(0);
	}

	public static int getFieldInitializedAutomatically() {
		return OnlyStatic.fieldInitializedAutomatically;
	}

	public static int getFieldInitializedBeforeConstructor() {
		return OnlyStatic.fieldInitializedBeforeConstructor;
	}

	public static int getPrivateField() {
		return OnlyStatic.privateField;
	}

	public static int getProtectedField() {
		return OnlyStatic.protectedField;
	}

	public static int getStandardField() {
		return OnlyStatic.standardField;
	}

	public static int getPublicField() {
		return OnlyStatic.publicField;
	}

	public static void testContextFields() {
		System.out.printf("%s\n", OnlyStatic.privateField);
		System.out.printf("%s\n", OnlyStatic.protectedField);
		System.out.printf("%s\n", OnlyStatic.standardField);
		System.out.printf("%s\n", OnlyStatic.publicField);
	}

	public static void testNoContextFields() {
		System.out.printf("%s\n", privateField);
		System.out.printf("%s\n", protectedField);
		System.out.printf("%s\n", standardField);
		System.out.printf("%s\n", publicField);
	}

	private static void testPrivateMethod() {
		System.out.printf("Private\n");
	}

	protected static void testProtectedMethod() {
		System.out.printf("Protected\n");
	}

	static void testStandardMethod() {
		System.out.printf("Standard\n");
	}

	public static void testPublicMethod() {
		System.out.printf("Public\n");
	}

	public static void test() {
		// 0 0 0 0
		System.out.printf("%s\n", OnlyStatic.getPrivateField());
		System.out.printf("%s\n", OnlyStatic.getProtectedField());
		System.out.printf("%s\n", OnlyStatic.getStandardField());
		System.out.printf("%s\n", OnlyStatic.getPublicField());
		// 0 0 0 0
		System.out.printf("%s\n", OnlyStatic.privateField);
		System.out.printf("%s\n", OnlyStatic.protectedField);
		System.out.printf("%s\n", OnlyStatic.standardField);
		System.out.printf("%s\n", OnlyStatic.publicField);

		// 0 0 0 0
		OnlyStatic.testContextFields();
		// 0 0 0 0
		OnlyStatic.testNoContextFields();

		// Private Protected Standard Public
		OnlyStatic.testPrivateMethod();
		OnlyStatic.testProtectedMethod();
		OnlyStatic.testStandardMethod();
		OnlyStatic.testPublicMethod();
	}

}
